lib/hashes.md

```diff
@@ -5,40 +5,40 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is a Hash?
 
-Replace me with your response...
+A hash is a collection of key-value pairs.
 
 # What are some examples of information that would be Hashes as opposed to some other data type?
 
-Replace me with your response...
+Some information that would be stored in a hash would be dictionary definitions, menu items and their prices, as well as names and ages.
 
 # How are Hashes and Arrays similar? How are they different?
 
-Replace me with your response...
+Hashes and arrays are similar in that they both store information that can be retrieved. They are different in that elements in an array are only associated with their numerical slot where a value in a hash can be accessed by calling its key.
 
 # How do you retrieve a particular value from a Hash?
 
-Replace me with your response...
+You can retrieve a particular value from a hash by calling it's associated key. If the key "Chris" is associated to the value "29", in the hash "names", I could retrieve the value, 29, with names["Chris"].
 
 # How do you add information to a Hash?
 
-Replace me with your response...
+You can add information to a hash with the "bracket-equals" method: names["Joe"]= "28"
 
 # How would you perform an operation on every element inside a Hash?
 
-Replace me with your response...
+You can perform an operation on every element in a hash using the method "each", which unlike an array, will pass two values (key and value) into the hash.
```

Comment:
I just want to nitpick a little here. You wrote that the `each` method "will pass two values (key and value) into the **hash**". But the hash is already set–you're operating on it. The term for the area where you define the operations to do is the "block". Take the below example:

```ruby
menu = {"Duck" => 500, "Chicken" => 120}

menu.each do |key, value|
  puts "The #{key} costs #{value}"
end
```

The part beginning with `do` and ending with `end` is "the block".

Please think about this for a moment and see if it makes sense. Ask questions if they occur to you.